File: examples/src/Text/RuleBasedFontsReplacement.cs
import aspose.pydrawing as drawing
import aspose.slides as slides

/*
This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Slides for .NET API reference 
when the project is build. Please check https:#docs.nuget.org/consume/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Slides for .NET API from http:#www.aspose.com/downloads, 
install it and then add its reference to this project. For any issues, questions or suggestions 
please feel free to contact us using http:#www.aspose.com/community/forums/default.aspx
*/

namespace Aspose.slides.Examples.CSharp.text
{
    class RuleBasedFontsReplacement
    {
        public static void Run()
        {
            #ExStart:RuleBasedFontsReplacement
            # The path to the documents directory.
            dataDir = RunExamples.GetDataDir_Text()

            # Load presentation
            Presentation presentation = new Presentation(dataDir + "Fonts.pptx")

            # Load source font to be replaced
            IFontData sourceFont = slides.FontData("SomeRareFont")

            # Load the replacing font
            IFontData destFont = slides.FontData("Arial")

            # Add font rule for font replacement
            IFontSubstRule fontSubstRule = new FontSubstRule(sourceFont, destFont, FontSubstCondition.WhenInaccessible)

            # Add rule to font substitute rules collection
            IFontSubstRuleCollection fontSubstRuleCollection = new FontSubstRuleCollection()
            fontSubstRuleCollection.add(fontSubstRule)

            # Add font rule collection to rule list
            presentation.FontsManager.FontSubstRuleList = fontSubstRuleCollection

            # Arial font will be used instead of SomeRareFont when inaccessible
            bmp = presentation.slides[0].get_thumbnail(1f, 1f)

            # Save the image to disk in JPEG format
            bmp.save(dataDir + "Thumbnail_out.jpg", System.Drawing.Imaging.ImageFormat.Jpeg)
            #ExEnd:RuleBasedFontsReplacement
        }
    }
}

File: examples/src/text/rule_based_fonts_replacement.py
import aspose.slides as slides


def rule_based_fonts_replacement(global_opts):
    # Load presentation
    with slides.Presentation(global_opts.data_dir + "text_fonts.pptx") as presentation:
        # Load source font to be replaced
        source_font = slides.FontData("SomeRareFont")

        # Load the replacing font
        dest_font = slides.FontData("Arial")

        # Add font rule for font replacement
        font_subst_rule = slides.FontSubstRule(source_font, dest_font, slides.FontSubstCondition.WHEN_INACCESSIBLE)

        # Add rule to font substitute rules collection
        font_subst_rule_collection = slides.FontSubstRuleCollection()
        font_subst_rule_collection.add(font_subst_rule)

        # Add font rule collection to rule list
        presentation.fonts_manager.font_subst_rule_list = font_subst_rule_collection

        # Arial font will be used instead of SomeRareFont when inaccessible
        img = presentation.slides[0].get_image(1, 1)

        # Save the image to disk in JPEG format
        img.save(global_opts.out_dir + "text_rule_based_font_replacement_out.jpg", slides.ImageFormat.JPEG)
